File: getting-started-mocks/02-first-useful-mock/node/notifier/notifier.test.js
import assert from 'node:assert/strict'
import test from 'node:test'

import { Notifier } from './notifier.js'

test('creates a notification when the package is delayed', async () => {
  const delayed = async () => 'delayed'
  const notifier = new Notifier(delayed)

  assert.equal(await notifier.notify('TRACK-123'), 'Package TRACK-123 is delayed')
})

test('creates no notification when the package is delivered', async () => {
  const delivered = async () => 'delivered'
  const notifier = new Notifier(delivered)

  assert.equal(await notifier.notify('TRACK-123'), null)
})

File: getting-started-mocks/02-first-useful-mock/go/notifier/notifier_test.go
package notifier

import "testing"

func TestNotifyDelayedPackage(t *testing.T) {
	delayed := func(string) (string, error) { return "delayed", nil }
	n := New(delayed)

	message, err := n.Notify("TRACK-123")
	if err != nil {
		t.Fatal(err)
	}
	if message != "Package TRACK-123 is delayed" {
		t.Fatalf("unexpected message: %q", message)
	}
}

func TestNotifyDeliveredPackage(t *testing.T) {
	delivered := func(string) (string, error) { return "delivered", nil }
	n := New(delivered)

	message, err := n.Notify("TRACK-123")
	if err != nil {
		t.Fatal(err)
	}
	if message != "" {
		t.Fatalf("expected no message, got %q", message)
	}
}
